src/cmd-buildupload

```diff
@@ -98,56 +110,68 @@ def s3_upload_build(args, builddir, dest):
                 args.enable_gz_peel):
             nogz = bn[:-3]
             img['path'] = nogz
-            s3_path = f'{dest}/{nogz}'
+            s3_path = f'{prefix}/{nogz}'
             set_content_disposition = True
 
         if not os.path.exists(path):
-            if s3_check_exists(args, s3_path):
+            if s3_check_exists(bucket, s3_path):
                 continue
             else:
                 raise Exception(f"{path} not found locally or in the s3 destination!")
 
+        extra_args = {}
         if set_content_disposition:
-            s3_cp(args, CACHE_MAX_AGE_ARTIFACT, path, s3_path,
-                  '--content-encoding=gzip',
-                  f'--content-disposition=inline; filename={img["path"]}')
-        else:
-            s3_cp(args, CACHE_MAX_AGE_ARTIFACT, path, s3_path)
+            extra_args = {
+                'ContentEncoding': 'gzip',
+                'ContentDisposition': f'inline; filename={img["path"]}'
+            }
+        s3_copy(path, bucket, s3_path, CACHE_MAX_AGE_ARTIFACT, args.acl, extra_args=extra_args)
         uploaded.add(bn)
 
     for f in os.listdir(builddir):
         # we do meta.json right after
         if f in uploaded or f == 'meta.json':
             continue
         path = os.path.join(builddir, f)
-        s3_cp(args, CACHE_MAX_AGE_ARTIFACT, path, f'{dest}/{f}')
+        s3_copy(path, bucket, f'{prefix}/{f}', CACHE_MAX_AGE_ARTIFACT, args.acl)
 
     # Now upload a modified version of the meta.json which has the fixed
     # filenames without the .gz suffixes. We don't want to modify the local
     # build dir.
     with tempfile.NamedTemporaryFile('w') as f:
         json.dump(build, f, indent=4)
         f.flush()
-        s3_cp(args, CACHE_MAX_AGE_METADATA, f.name, f'{dest}/meta.json',
-              '--content-type=application/json')
-
-
-def s3_check_exists(args, path):
-    path = f'{args.url}/{path}'
-    bucket, key = path.split("/", 1)
-    s3_args = ['aws', 's3api', 'head-object', '--bucket', bucket, '--key', key]
-    return subprocess.call(s3_args, stdout=subprocess.DEVNULL) == 0
-
-
-def s3_cp(args, max_age, src, dest, *s3_args):
-    acl = f'--acl={args.acl}'
-    max_age = f'--cache-control=max-age={max_age}'
-    dest = f's3://{args.url}/{dest}'
-    s3_args = ['aws', 's3', 'cp', acl, src, dest, max_age, *s3_args]
-    print("%s: %s" % ("Would run" if args.dry_run else "Running",
-                      subprocess.list2cmdline(s3_args)))
-    if not args.dry_run:
```

Comment:
We lost `dry_run` handling here. Maybe we could aggregate the args in a dict, then call `s3.upload_file(**args))` and in the dry run mode just print the dict?

Reply:
Right, good catch, fixed that

Reply:
>Maybe we could aggregate the args in a dict

I wanted to explicitly make argparse args and `s3_copy` args different so that there would be less confusion - perhaps s3_copy could be reused later